File: angularapp/src/app/models/assignment.ts

export interface Assignment{
    id:number;
    courseId:number;
    instructorId:number;
    assignmentName:string;
    postedDate:Date;
    dueDate:Date;
}

export class AssignmentQuestion{
    id:number=0;
    assignmentId:number=0;
    question:string='';
    options:string[]=[];
    answer:string='';
    constructor(question:string,options:string[],answer:string){
        this.question=question;
        this.options=options;
        this.answer=answer
    }
}

export interface CompletedAssignmentData{
    id:number;
    assignmentId:number;
    instructorId:number;
    courseId:number;
    userId:number;
    marksObtained:number;
    assignmentName:string;
}

File: springapp/src/main/java/com/example/springapp/model/Assignment.java
package com.example.springapp.model;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Assignment {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;
    private Long instructorId;
    private Long courseId;
    private String assignmentName;
    private Date postedDate;
    private Date dueDate;
}
